src/appleseed.python/bindblendermeshconverter.cpp

```diff
@@ -0,0 +1,222 @@
+//
+// Copyright (c) 2018 Jonathan Dent, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.renderer headers.
+#include "renderer/api/object.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/python.h"
+
+namespace bpy = boost::python;
+using namespace renderer;
+
+// Blender data structures
+struct MeshFace
+{
+    unsigned int v[4];
+    short mat_nr;
+    char edcode, flag;
+};
+
+struct MeshVert
+{
+    float co[3];
+    short no[3];
+    char flag, bweight;
+};
+
+struct MeshTexFace
+{
+    float uv[4][2];
+    void *tpage;
+    char flag, transp;
+    short mode, tile, unwrap;
+};
+
+
+// The following function takes a series of pointers to Blender mesh data and modifies the appleseed MeshObject entity
+//      blender_object = the appleseed MeshObject created earlier in the export process
+//      vertices_length = the number of vertices in the mesh
+//      vertex_pointer = a string pointer to the first element of the vertex array
+//      Same template applies to mesh faces
+
+void convert_bl_mesh(
+    MeshObject*                 blender_mesh,
+    const size_t                vertices_length,
+    const size_t                vertex_pointer,
+    const size_t                triangle_length,
+    const size_t                triangle_pointer,
+    const size_t                uv_layer_pointer,
+    const bool                  export_normals,
+    const bool                  export_uvs)
+{
+    // Convert size_t numbers to actual pointers
+    MeshVert* vertices = reinterpret_cast<MeshVert *>(vertex_pointer);
+    MeshFace* triangles = reinterpret_cast<MeshFace *>(triangle_pointer);
+    MeshTexFace* uv_face = reinterpret_cast<MeshTexFace *>(uv_layer_pointer);
+
+    blender_mesh->reserve_vertices(vertices_length);
+    blender_mesh->reserve_triangles(triangle_length);
+
+    if (export_normals == true)
+    {
```

Comment:
In order to increase vertical code density, we don't use curly braces for one-line `if` bodies. Although this is completely correct and sometimes even a recommended practice, for the sake of consistency I'd suggest to remove them.

We _do_ however require curly braces for multi-line `if` bodies, even if those multiple lines form a single statement, e.g.

BAD:

    if (condition)
        function_call(
            long_arg1,
            long_arg2,
            long_arg3);

GOOD:

    if (condition)
    {
        function_call(
            long_arg1,
            long_arg2,
            long_arg3);
    }